### ISSUE TITLE
Validate website through previews generated by CI

### ISSUE THREAD

**Issue by gberche-orange** [cloudfoundry-community/terraform-provider-cloudfoundry#112]
We need to validate that the website is correct and does not include broken links, especially after #85 

We would need a documentation preview to be generated by the travis-ci build and published somewhere for manual inspection (e.g. through github pages https://pages.github.com/ )

More details about previewing providers website doc into https://github.com/hashicorp/terraform-website#previewing-changes-from-the-provider-or-core-repositories

**Comment by lixilin2301** [comment]
Can we use makefile `website-test` to validate all links by code and use `website` to generate local server for manual check, just like other providers? e.g. [Kubernetes](https://github.com/terraform-providers/terraform-provider-kubernetes/blob/master/GNUmakefile#L48).

**Comment by gberche-orange** [comment]
thanks @lixilin2301 Would you be able to submit a PR inspired from the K8S provider automation ?

**Comment by lixilin2301** [comment]
Yes, sure, I can do it. At the same time, we also need to make a pr to [hashcorp/terraform-website](https://github.com/hashicorp/terraform-website/tree/master) to integrate our website to their page, i.e. some symbolic links.

**Comment by lixilin2301** [comment]
@gberche-orange , I see we have quite a few branches for websites, e.g. dev(some fresh changes), dev-website and stable-website (some old, last change 1 year ago), do you have an idea which branch I could use? In addition, since website-test is run by hashcorp/terraform-website, we need to create some symbolic links to their repo. Otherwise, once we create the travis-ci build, it fails. 

And, @mevansam or @gberche-orange maybe, I see you changed layout name from 'cloudfoundry.erb' to 'cf.erb', do you have a specific reason for it?  Since I observe some sublinks(https://github.com/mevansam/terraform-provider-cf/blob/dev/website/docs/d/asg.html.markdown) still use 'cloudfoundry' layout which is certainly broken. To make it consistent, I suggest we keep using 'cloudfoundry' everywhere.

**Comment by gberche-orange** [comment]
thanks @lixilin2301 

> I see we have quite a few branches for websites, e.g. dev(some fresh changes), dev-website and stable-website (some old, last change 1 year ago), do you have an idea which branch I could use?

the dev branch is the one that is up to date. It will be merged into master on the next release.

>  I see you changed layout name from 'cloudfoundry.erb' to 'cf.erb', do you have a specific reason for it?

The changes to the layout references from "cf" to "cloudfoundry" in the markdown files was likely initiated in #85. Lets confirm this breaks and then proceed if need with renaming `cf.erb` in `cloudfoundry.erb` along with cross-references if any.

> To make it consistent, I suggest we keep using 'cloudfoundry' everywhere.

+1

**Comment by gberche-orange** [comment]
Two pending tasks:
 * [ ] add symbolic links of our layout and doc folder to terraform-website. Pending merge of https://github.com/hashicorp/terraform-website/pull/477
* [ ] add make website-test to travis-ci.

**Comment by loafoe** [comment]
@gberche-orange @lixilin2301 @ArthurHlt is this issue still current now that the provider is published to https://registry.terraform.io/providers/cloudfoundry-community/cloudfoundry/latest ?

**Comment by gberche-orange** [comment]
@loafoe I believe this issue suggests to refine automatic tests to detect regressions to website rendering before publishing.

Even though website is currently published, detecting regression before release might still be interesting

**Comment by loafoe** [comment]
@gberche-orange added linter Github Action for the Markdown file in 75f2f3d
